sphinx_needs/directives/needuml.py

```diff
@@ -171,6 +196,18 @@ def process_needuml(app, doctree, fromdocname):
         id = node.attributes["ids"][0]
         current_needuml = env.needs_all_needumls[id]
 
+        parent_need_id = None
+        # Check if current needuml is needarch
+        if current_needuml["is_arch"]:
+            # Check if needarch is only used inside a need
+            from sphinx_needs.directives.need import Need
```

Comment:
This import should be at the top of the file.
It does not save really any time, if done in this if-statement.